app/components/address-search.js

```diff
@@ -0,0 +1,66 @@
+import Component from '@ember/component';
+import { action } from '@ember-decorators/object';
+// import { argument } from '@ember-decorators/argument';
+
+export const geocodedLayer = {
+  type: 'circle',
+  paint: {
+    'circle-radius': {
+      stops: [
+        [
+          10,
+          5,
+        ],
+        [
+          17,
+          12,
+        ],
+      ],
+    },
+    'circle-color': 'rgba(199, 92, 92, 1)',
+    'circle-stroke-width': {
+      stops: [
+        [
+          10,
+          20,
+        ],
+        [
+          17,
+          18,
+        ],
+      ],
+    },
+    'circle-stroke-color': 'rgba(65, 73, 255, 1)',
+    'circle-opacity': 0,
+    'circle-stroke-opacity': 0.2,
+  },
+};
+
+export default class AddressSearch extends Component {
+  // @argument
+  map;
+
+  // @argument
+  onSelectSearchResult = () => {}
```

Comment:
Can you add comments for `onSelectSearchResult` and `onClearSearchResult` to explain their expected behavior?

Reply:
@trbmcginnis sure, full comments forthcoming. The front end doesn't do any of the filtering, the backend does. All the front end is responsible for is sending point coordinates and radius distance to the backend, and providing visual feedback to users.